### PR TITLE
[ELY-1391] Check if all elements in elytron-1_0.xsd are parsed correctly in ElytronXmlParser.

### PR DESCRIPTION
**Pull request by ivassile**
wildfly-elytron: https://issues.jboss.org/browse/ELY-1391
JBEAP: https://issues.jboss.org/browse/JBEAP-13416

### CODE REVIEW

File: src/main/resources/schema/elytron-1_0.xsd [hunk 1]
```diff
@@ -209,7 +209,6 @@
         <xsd:choice minOccurs="0" maxOccurs="1">
             <xsd:element name="key-store-credential" type="key-store-ref-type" minOccurs="1" maxOccurs="1"/>
             <xsd:element name="key-store-clear-password" type="clear-password-type" minOccurs="1" maxOccurs="1"/>
-            <xsd:element name="credential-store" type="credential-store-reference-type" minOccurs="1" maxOccurs="1"/>
```
Comment (darranl) [inline src/main/resources/schema/elytron-1_0.xsd]: Don't worry about the credential store ones they have been fixed in another PR.

File: src/main/resources/schema/elytron-client-1_1.xsd [hunk 2]
```diff
@@ -210,7 +210,6 @@
         <xsd:choice minOccurs="0" maxOccurs="1">
             <xsd:element name="key-store-credential" type="key-store-ref-type" minOccurs="1" maxOccurs="1"/>
             <xsd:element name="key-store-clear-password" type="clear-password-type" minOccurs="1" maxOccurs="1"/>
-            <xsd:element name="credential-store" type="credential-store-reference-type" minOccurs="1" maxOccurs="1"/>
```
Comment (darranl) [inline src/main/resources/schema/elytron-client-1_1.xsd]: Again these are fixed in another PR.

File: src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java [hunk 3]
```diff
@@ -1259,7 +1259,7 @@ static void parseKeyStoreType(ConfigurationXMLStreamReader reader, final Version
                         resourceSource = parseNameType(reader);
                         break;
                     }
-                    case "uri": {
+                    case "load-from": {
```
Comment (darranl) [inline src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java]: For the purposes of backwards compatibility the implementation in the parser tends to win over the the schema definition.

A configuration written correctly against the schema would not work so we know that know users can possibly be using it, however if they are aware of the discrepency they can create a config based on the parser implementation.

Versions 1.0 and 1.0.1 of the schema should be updated to reflect reality, i.e. the schema should be updated to the element is called 'uri'.

We could then choose to correct this in version 1.1 of the schema so uri would be supported if (!xmlVersion.isAtLeast(VERSION_1_1)) whilst load-from would then be supported if (xmlVersion.isAtLeast(VERSION_1_1)) alternatively we could support both forms in 1.1 and add some documentation in the schema to say the uri form is deprecated.

**Comment by wolfc** [comment]
Keep in mind that xsd can not be changed after a component release. It is considered golden for usage in other tools (like JBDS) which only download and cache it once.

**Comment by darranl** [comment]
I say don't break existing config in use by end users is the no 1 rule that trumps all subsequent rules, and schema changes should only be to fix true bugs in the schema and not to enhance the schema.

**Comment by honza889** [comment]
@ivassile what about "credential-store-reference" in "key-store-type" ? looks like it is in two XSD (1_0_1 and 1_1) but not supported by parser.

**Comment by darranl** [comment]
@honza889 those have already been cleaned up.

**Pull request merged**